### PR TITLE
Fix inferDefinitions and add test.

### LINKED ISSUE #107: inferDefinitions should support dictionnary as well.
[Tested on 0.6.0 since the 0.6.3 does not work for me at the moment because of issue [91](https://github.com/andyglow/scala-jsonschema/issues/91)] 
The given code should work:

```

case class FooX(a: Int)
case class FooY(b: Int)

class AsDraft07Spec extends AnyWordSpec {

  "AsValue.schema" should {

    "emit Object" in {
      import `object`.Field


      val o = `object`(
        Field("foo", `string`[String](None, None)),
        Field("bar", `integer`, required = false),
        Field("baz", `ref`[Boolean]("scala.Boolean", `boolean`("my-bool"))),
        Field("mapObject", `dictionary`[String, FooX, Map](`ref`[FooX]("FooX", `object`(Field("a", `integer`))))),
        Field("listObject", `array`[FooY, List](`ref`[FooY]("FooY", `object`(Field("b", `integer`)))))
      )

      val expected = obj(
        f"$$schema" -> "http://json-schema.org/draft-07/schema#",
        f"$$id" -> "http://example.com/foobarbaz.json",
        "type" -> "object",
        "additionalProperties" -> false,
        "required" -> arr("listObject", "baz", "mapObject", "foo"),
        "properties" -> obj(
          "foo" -> obj("type" -> "string"),
          "bar" -> obj("type" -> "integer"),
          "baz" -> obj(f"$$ref" -> "#my-bool"),
          "mapObject" -> obj(
            "type" -> "object",
            "patternProperties" -> obj(
              "^.*$" -> obj(
                f"$$ref" -> "#FooX"))
          ),
          "listObject" -> obj(
            "type" -> "array",
            "items" -> obj(f"$$ref" -> "#FooY")
          )
        ),
        "definitions" -> obj(
          "my-bool" -> obj(
            f"$$id" -> "#my-bool",
            "type" -> "boolean"),
          "FooX" -> obj(
            f"$$id" -> "#FooX",
            "type" -> "object",
            "additionalProperties" -> false,
            "properties" -> obj("a" -> obj("type" -> "integer")),
            "required" -> arr("a")
          ),
          "FooY" -> obj(
            f"$$id" -> "#FooY",
            "type" -> "object",
            "additionalProperties" -> false,
            "properties" -> obj("b" -> obj("type" -> "integer")),
            "required" -> arr("b")
          )
        ))

      AsValue.schema(o, Draft07(id = "http://example.com/foobarbaz.json")) should containJson(expected)
    }
  }
}
````

I think the only thing necessary to make it work is to add a case for dictionnary in inferDefinitions

### PR DESCRIPTION
**Pull request by michellemay**
fixes #107

### CODE REVIEW

File: core/src/main/scala/com/github/andyglow/json/JsonFormatter.scala [hunk 1]
```diff
@@ -63,4 +63,4 @@ class JsonFormatter(step: Int = 2) {
   }
 }
 
-object JsonFormatter extends JsonFormatter(step = 2)
\ No newline at end of file
+object JsonFormatter extends JsonFormatter(step = 2)
```
Comment (andyglow) [inline core/src/main/scala/com/github/andyglow/json/JsonFormatter.scala:66]: should it be included?

File: api/src/test/scala/com/github/andyglow/jsonschema/AsDraft07Spec.scala [hunk 2]
```diff
@@ -7,6 +7,8 @@ import json.schema.Version._
 import org.scalatest.matchers.should.Matchers._
 import org.scalatest.wordspec.AnyWordSpec
 
+case class FooX(a: Int)
+case class FooY(b: Int)
```
Comment (andyglow) [inline api/src/test/scala/com/github/andyglow/jsonschema/AsDraft07Spec.scala:11]: can we have these guys moved into spec's companion object?

**Review by andyglow — CHANGES_REQUESTED** [review]

File: core/src/main/scala/com/github/andyglow/json/JsonFormatter.scala (hunk 1 above)
Comment (michellemay) [reply core/src/main/scala/com/github/andyglow/json/JsonFormatter.scala:66]: it's only a matter of intellij adding a newline automatically..

Comment (michellemay) [reply core/src/main/scala/com/github/andyglow/json/JsonFormatter.scala]: reverted

File: api/src/test/scala/com/github/andyglow/jsonschema/AsDraft07Spec.scala (hunk 2 above)
Comment (michellemay) [reply api/src/test/scala/com/github/andyglow/jsonschema/AsDraft07Spec.scala]: done

**Pull request merged**